pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -226,6 +226,23 @@ func (r *ReconcileLogCollector) Reconcile(request reconcile.Request) (reconcile.
 		}
 	}
 
+	var splunkCredential *render.SplunkCredential
+	if instance.Spec.AdditionalStores != nil {
+		if instance.Spec.AdditionalStores.Splunk != nil {
+			splunkCredential, err = getSplunkCredential(r.client)
```

Comment:
We need a watch for this in the `add` function above to ensure if this changes the controller will re-Reconcile.

Reply:
Done, I added a watch along with the elastic secrets